### PR TITLE
My thought on the  Electric bill project

### PR DESCRIPTION
**Pull request by Ngozistephen**

### CODE REVIEW

**Comment by Ngozistephen** [comment]
Thought on the chessboard project

**Review by ahmadinjo — COMMENTED** [review]
This is brilliant bro. Well done. Team lead should approve it whenever he is available. Thanks, really appreciate.

**Comment by Ngozistephen** [comment]
Thank you sir. we move 👍

**Review by ahmadinjo — CHANGES_REQUESTED** [review]
- Bugs 🐛within the code that leads to calculation error for units above 150.
- Add some form of limit to the number of units a client can enter (i.e. show error for negative values). You can also restrict them from inputting negative values using the "min" attribute on the input element.
- You can add the Naira (₦) symbol to the total amount.

File: stephenthought.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			$temp = 50 * $first_unit_cost;
+			$remaining_units = $units - 50;
+			$bill = $temp + ($remaining_units * $second_unit_cost);
+		}
+
+        // Else if, check that units consumed is less than equal to the 250, if yes then total electricity bill will be:
+
+        // Total Electricity Bill = (50* 3.5) + (50 * 4.00) + (units- 150) * 5.20
+		else if($units > 150 && $units <= 250) {
+			$temp = (50 * $first_unit_cost) + (50 * $second_unit_cost);
```
Comment (ahmadinjo) [inline stephenthought.php]: NB: First 50 units at 3.50, next 100 at 4.00, next 100 at 5.20, remaining at 6.50
Check the project requirement carefully again. You can use my code as guide.

File: stephenthought.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			$remaining_units = $units - 150;
+			$bill = $temp + ($remaining_units * $third_unit_cost);
+		}
+
+        // Else check that units consumed greater than 250, if yes then total electricity bill will be:
+
+        // Total Electricity Bill = (50 * 3.5) + (50 * 4.00) + (50 * 5.20) + (units- 250) * 6.50
+        else {
+			$temp = (50 * $first_unit_cost) + (50 * $second_unit_cost) + (50 * $third_unit_cost);
```
Comment (ahmadinjo) [inline stephenthought.php]: Same thing above applies to this. Just take your time to go through the project requirements

File: stephenthought.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		}
+		return number_format((float)$bill, 2, '.', '');
+	}
+	?>
+	<div class="container">
+		<h1>Capstone Project Calculating Electric bill in PHP</h1>
+		<div class="form-group">
+		<form action="" method="post">
+			<div class="form-group">
+				<input type="number" name="kwh" placeholder="Please enter no. of Units" class="form-control" />
```
Comment (ahmadinjo) [inline stephenthought.php]: Don't forget to add the restriction on the minimum units the client can enter

**Comment by Kansoldev** [comment]
@Ngozistephen, you have done very well, thank you for your contribution. For the chessboard project, i would merge your pull request to the main repo when you have fixed the issues that the electricity bill project has. For the electricity bill, change the file name to the one that the repo has and make sure you apply the minor bug fixes that @ahmadinjo noticed because they are important. After that you can submit your pull request again

**Review by Kansoldev — DISMISSED** [review]
Code accepted for chessboard project

**Comment by Ngozistephen** [comment]
okey i will

**Comment by Ngozistephen** [comment]
pls concerning the bug  have checked is working fine

**Review by ahmadinjo — CHANGES_REQUESTED** [review]
You've not effected the other changes I pointed out to you. Check the logic on the base master and compare it to yours. You can also implement the logic in your code too since it's supposed to be a team work.
To see the bug 🐛 I'm talking about, test 150 units and 151 units and note the difference.

**Comment by Kansoldev** [comment]
The issue has been solved and i have merged @ahmadinjo pull request

**Comment by Ngozistephen** [comment]
okey that's good

**Pull request closed without merging**